components/TaskList.tsx

```diff
@@ -1,37 +1,41 @@
 import { FlatList, View, Text, StyleSheet, Pressable } from "react-native";
 import { useSQLiteContext } from "expo-sqlite";
-import { useEffect, useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import TaskItem from "./TaskItem";
 import { Task } from "../models/tasks";
 import { AntDesign } from "@expo/vector-icons";
+import { TasksContext } from "../store/tasks-context";
 
-export default function TaskList({ cat }) {
-  const [tasks, setTasks] = useState<Task[]>([]);
-  const [isDisplayed, setIsDisplayed] = useState(false);
+type Cat = {
+  category: string;
+};
 
-  const db = useSQLiteContext();
+export default function TaskList({ cat }: { cat: Cat }) {
+  const TaskCxt = useContext(TasksContext);
```

Comment:
I suggest define a custom hook to consume a context, with definition in form of:
```
function useTaskContext(){
    return useContext(TasksContext)
}
```

Define this inside the same file that has Context provider.

So the consumer of your context provider need only import the hook `useTaskContext()` and need not import both useContext as well as the correct context.